### DISCUSSION TITLE
How to use with Unity's AuthenticationService?

### DISCUSSION THREAD

**Discussion by jan** [Facepunch/Facepunch.Steamworks#734] [Q&A]
I am implementing authentication via Unity's AuthenticationService and multiple providers.

For Steam, I need a ticket to pass into
```c#
AuthenticationService.Instance.SignInWithSteamAsync(string sessionTicket)
```

As I founsd this quite cumbersome using SteamWorks.NET, I am now trying Facepunch instead.

But I can't seem to get the right ticket string. I'm trying:

```c#
AuthTicket ticket = SteamUser.GetAuthSessionTicket();

string sessionTicketA = BitConverter.ToString(ticket.Data);
AuthenticationService.Instance.SignInWithSteamAsync(sessionTicketA);
// => Unity.Services.Authentication.AuthenticationException: invalid token ---> Unity.Services.Authentication.WebRequestException: {"title":"PERMISSION_DENIED","detail":"invalid token","details":[],"status":401}

// In the manual, they remove dashes, so let's try that instead
// https://docs.unity.com/ugs/en-us/manual/authentication/manual/platform-signin-steam
string sessionTicketB = sessionTicketA.Replace("-", string.Empty);
AuthenticationService.Instance.SignInWithSteamAsync(sessionTicketB);
// => Unity.Services.Authentication.AuthenticationException: invalid token ---> Unity.Services.Authentication.WebRequestException: {"title":"PERMISSION_DENIED","detail":"invalid token","details":[],"status":401}
```

Is Facepunch's `SteamUser.GetAuthSessionTicket` even supposed to be a replacement for SteamWorks.NET's `SteamUser.GetAuthTicketForWebApi`?

Should this work? Is it actually a permission issue? I've tried to double check that...

Thanks for your help.

**Comment by TomaszBlaszczyk** [comment]
I have the same issue. Do you already know how to fix it?

**Reply by jan** [reply]
No, I gave up and stopped using Facepunch. I think it's an abandoned project.
